### PR TITLE
Major enhancements to cli

### PR DESCRIPTION
**Pull request by kbacha**
Several things went into this change. Some was refactoring and some were
new enhancements.

*Added All Assets*

The first change was the most straight forward, the assets endpoint was
added. This end point features filtering via the command line args of
code and issuer.

*Switched to using the Iter*

The second change was the migration to using the iterator. This majorly
simplifies the code and eliminates the need for the recursion that was
there to begin with. I use the `enumerate` iterator to pair each line
with the index so that we can paginate.

*Functions return result types now*

Instead of duplicating the exception handling through out, instead we
just have each function return the client errors and handle it all at
the end.

*Choose the network!*

Another option that was added was the ability specify which network you
wished to join!

*Move the groups of functions to modules*

The file was feeling a little complex so I moved each group of commands
to their own module file. They were already modules so this was very
straightforward.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -39,128 +59,91 @@ fn build_app<'a, 'b>() -> App<'a, 'b> {
...
-                ::std::process::exit(1);
-            }
-            err => {
-                eprintln!("An unknown error occurred: {:?}", err);
-                ::std::process::exit(1);
-            }
-        }
-    }
+    let client = if let Some(host) = matches.value_of("host") {
+        Client::new(&host).expect("Failed to initialize client")
```
Comment (correlator) [inline cli/src/main.rs]: 🔥

File: cli/src/main.rs [hunk 2]
```diff
@@ -3,13 +3,33 @@ extern crate stellar_client;
...
         .version("0.1")
         .about("Access the stellar horizon API via the command line.")
         .setting(AppSettings::SubcommandRequired)
+        .arg(
+            Arg::with_name("host")
+                .takes_value(true)
+                .short("h")
+                .long("host")
+                .conflicts_with_all(&["test-net", "pub-net"])
+                .help("The host url to attempt to connect to"),
```
Comment (correlator) [inline cli/src/main.rs]: Is it worth mentioning that the default is test-net here as well.

File: cli/src/account.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    println!("ID:       {}", account.id());
+    println!("Sequence: {}", account.sequence());
+
+    Ok(())
+}
+
+pub fn transactions<'a>(client: Client, matches: &'a ArgMatches) -> Result<()> {
+    let id = matches.value_of("ID").expect("ID is required");
+    let endpoint = account::Transactions::new(id).order(Order::Desc);
```
Comment (correlator) [inline cli/src/account.rs]: Do we want limit, order, or cursor to be options?

File: cli/src/account.rs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    for (i, result) in iter.enumerate() {
+        let txn = result?;
+
+        println!("ID:             {}", txn.id());
+        println!("source account: {}", txn.source_account());
+        println!("created at:     {}", txn.created_at());
+        println!("");
+
+        if (i + 1) % 10 == 0 && !utils::next_page() {
```
Comment (correlator) [inline cli/src/account.rs]: This whole thing is so good.  Especially with the new iterators on Result.

File: cli/src/main.rs [hunk 5]
```diff
@@ -39,128 +59,91 @@ fn build_app<'a, 'b>() -> App<'a, 'b> {
...
+    } else if matches.is_present("pub-net") {
+        Client::horizon().unwrap()
+    } else {
+        Client::horizon_test().unwrap()
+    };
 
-    pub fn transactions<'a>(client: Client, matches: &'a ArgMatches, cursor: Option<String>) {
-        let id = matches.value_of("ID").expect("ID is required");
-        let mut endpoint = account::Transactions::new(id).order(Order::Desc);
+    // Master match block. All subcommands need to be captured here.
```
Comment (robertDurst) [inline cli/src/main.rs]: Wow match statements are awesome!

File: cli/src/account.rs (hunk 3 above)
Comment (kbacha) [reply cli/src/account.rs]: perhaps. Order makes sense, cursor could be a starts-from, and limit is page size (which may not control how many we actually fetch at a time). However, that can be another branch as I think this one contains enough.

**Pull request merged**